mne/io/ctf/ctf.py

```diff
@@ -149,6 +150,10 @@ def __init__(self, directory, system_clock='truncate', preload=False,
             last_samps=last_samps, filenames=fnames,
             raw_extras=raw_extras, orig_format='int', verbose=verbose)
 
+        # Add bad segments as Annotations (correct for start time)
+        s_time = -res4['pre_trig_pts'] / float(info['sfreq'])
+        self.annotations = _read_bad_segments(directory, s_time)
```

Comment:
Ideally this s_time would be the start of the 'raw.times' array. Is that possible with the current fif structures?

Reply:
`raw.times` always starts at `0.` but there is `raw.first_samp` which is an offset, is that what you mean?

Reply:
`raw.first_samp` at least from what I saw doesn't change `raw.times[0]` but does skip the first `raw.first_samp` samples of the data array. Is that the correct understanding?

Reply:
I'm not sure what it does in the context of CTF. But for Neuromag data it is the offset between when the acquisition started and the recording began. In MNE we use it when cropping files, too.

Reply:
The problem with setting raw.first_samp is that it skips those samples as stored on disk, so while it makes sense of the timing issues needs to say set to 0 to read in all of the data.

Reply:
first_samp should not impact the reading of the first samples on disk. It's basically an increment in events and allows to know when samples where recorded wrt the original absolute time

Reply:
That doesn't match what I am seeing.
for instance running this test code
```
import os.path as op
from mne.io import read_raw_ctf
from mne.datasets import testing

ctf_dir = op.join(testing.data_path(download=True), 'CTF')
ctf_fname = op.join(ctf_dir, 'testdata_ctf_short.ds')
raw = read_raw_ctf(ctf_fname, preload=True)

print ('first_samp - %d' % raw.first_samp)
print ('_data.shape', raw._data.shape)
print ('_data[3,0]', raw._data[3, 0])
print ('_data[3,5]', raw._data[3, 5])

print ('len(raw.times) - %d' % len(raw.times))
print ('raw.times.min() - %f' % raw.times.min())
print ('raw.times.max() - %f' % raw.times.max())
```

on the current master(*not this branch*) returns
```
first_samp - 0
('_data.shape', (356, 4801))
('_data[3,0]', 16442.588333333333)
('_data[3,5]', 16442.592499999999)
len(raw.times) - 4801
raw.times.min() - 0.000000
raw.times.max() - 4.000000
```

if I change 
https://github.com/mne-tools/mne-python/blob/da1d32978b6033dd89609f12af291ed619264c7d/mne/io/ctf/ctf.py#L146
to
```first_samps = [5] * len(last_samps)```

returns
```
first_samp - 5
('_data.shape', (356, 4796))
('_data[3,0]', 16442.592499999999)
('_data[3,5]', 16442.596666666665)
len(raw.times) - 4796
raw.times.min() - 0.000000
raw.times.max() - 3.995833
```

so it certainly seems like setting first_samp skips the opening of the data array.